grpc/server.go

```diff
@@ -71,9 +93,9 @@ func (sb *serverBuilder) Add(desc *grpc.ServiceDesc, impl any) *serverBuilder {
 // gracefully stop the server if one is caught, causing the start() function to
 // exit.
 func (sb *serverBuilder) Build(tlsConfig *tls.Config, statsRegistry prometheus.Registerer, clk clock.Clock) (func() error, error) {
-	// Add the health service to all servers.
-	healthSrv := health.NewServer()
-	sb = sb.Add(&healthpb.Health_ServiceDesc, healthSrv)
+	// Register the health service with the server.
+	sb.healthSrv = health.NewServer()
+	sb = sb.Add(&healthpb.Health_ServiceDesc, sb.healthSrv)
```

Comment:
This assignment isn't necessary. The `sb.Add()` method modifies `sb` in place. It only returns the result so that chaining (e.g. `.New().Add().Add().Build()`) works.

Reply:
That assignment predates my change, I'll go ahead and fix it now though.